File: src/twosum/twosum.js
"use strict";
exports.__esModule = true;
exports.main = void 0;
function twoSum(nums, target) {
    var map = new Map();
    for (var i = 0; i < nums.length; i++) {
        var complement = target - nums[i];
        if (map.has(complement)) {
            return [map.get(complement), i];
        }
        map.set(nums[i], i);
    }
    throw new Error("No two sum solution exists.");
}
function main() {
    var nums = [2, 7, 11, 15];
    var target = 9;
    try {
        var result = twoSum(nums, target);
        console.log("Indices: ".concat(result, ", Values: [").concat(nums[result[0]], ", ").concat(nums[result[1]], "]"));
    }
    catch (error) {
        console.error(error.toString());
    }
}
exports.main = main;
main();

File: src/twosum/two_sum.go
package main

func twoSum(nums []int, target int) []int {
	// Create a map to store the indices of the numbers we have seen so far
	indices := make(map[int]int)

	for i, num := range nums {
		// Calculate the complement needed to reach the target
		complement := target - num

		// Check if the complement exists in the map
		if index, found := indices[complement]; found {
			// Return the indices if we find the complement
			return []int{index, i}
		}

		// Store the current number's index in the map
		indices[num] = i
	}

	// Return nil if no solution is found
	return nil
}
